error: cannot convert 'typedef struct _result_string' to 'struct string'

### Describe the bug

can't compile a pure v code.

### Reproduction Steps

d.v

```v
module main

const x = c(c(c('123') or { '456' }) or { '444' }) or {'xx'}

fn a(s string)?string {
        if s == 'b' {
                return 'a'
        } else {
                return none
        }
}

fn b(s string)?string {
        if s == 'a' {
                return 'b'
        } else {
                return none
        }
}


fn c(s string)!string {
        return a(s) or { b(s) or {'c'}}
}

fn main() {
        println(x)
}
```

### Expected Behavior

It should be compiled.

### Current Behavior

```
$ v d.v
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/d.01JSVNXKVHS8ARBJ75HCTCEX67.tmp.c:5706: error: cannot convert 'typedef struct _result_string' to 'struct string'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

It maybe related to bug #24334

in file `vlib/v/gen/c/fn.v`,  in 
```v
fn (mut g Gen) call_expr(node ast.CallExpr) {
```

the `g.last_tmp_call_var` need to `pop` when `g.or_block(tmp_opt, node.or_block, node.return_type)`, otherwise , `g.last_tmp_call_var` will leak.

### V version

V 0.4.10 d9b808cdc3a0f711c07a7f3e5d302eb3250ae208.d9b808c

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.10 d9b808cdc3a0f711c07a7f3e5d302eb3250ae208.d9b808c
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 24.04.2 LTS
|Processor           |8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz
|Memory              |10.04GB/15.51GB
|                    |
|V executable        |/media/HD/github/kbkpbot/v/v
|V last modified time|2025-04-27 13:08:39
|                    |
|V home dir          |OK, value: /media/HD/github/kbkpbot/v
|VMODULES            |OK, value: /home/mars/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/mars/v/bug
|                    |
|Git version         |git version 2.43.0
|V git status        |weekly.2025.17-4-gd9b808cd
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|gcc version         |gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|clang version       |Ubuntu clang version 18.1.3 (1ubuntu1)
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.39-0ubuntu8.4) 2.39

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1048,7 +1048,7 @@ fn (mut g Gen) call_expr(node ast.CallExpr) {
 					g.write('\n ${cur_line}')
 				}
 			} else {
-				if !g.inside_or_block && g.last_tmp_call_var.len > 0 {
+				if !g.inside_or_block && g.last_tmp_call_var.len > 0 && !cur_line.contains(' = ') {
 					g.write('\n\t*(${unwrapped_styp}*)${g.last_tmp_call_var.pop()}.data = ${cur_line}(*(${unwrapped_styp}*)${tmp_opt}.data)')
 				} else {
 					g.write('\n ${cur_line}(*(${unwrapped_styp}*)${tmp_opt}.data)')

--- a/vlib/v/tests/fns/call_result_test.v
+++ b/vlib/v/tests/fns/call_result_test.v
@@ -1,0 +1,27 @@
+module main
+
+const x = c(c(c('123') or { '456' }) or { '444' }) or { 'xx' }
+
+fn a(s string) ?string {
+	if s == 'b' {
+		return 'a'
+	} else {
+		return none
+	}
+}
+
+fn b(s string) ?string {
+	if s == 'a' {
+		return 'b'
+	} else {
+		return none
+	}
+}
+
+fn c(s string) !string {
+	return a(s) or { b(s) or { 'c' } }
+}
+
+fn test_main() {
+	assert x == 'c'
+}

